internal/pkg/template/repository/manifest/records.go

```diff
@@ -156,3 +166,30 @@ func (v *TemplateRecord) DefaultVersionOrErr() (VersionRecord, error) {
 	}
 	return version, nil
 }
+
+func (v *TemplateRecord) CheckComponents(components *model.ComponentsMap) bool {
+	for _, component := range v.Requirements.Components {
+		if _, found := components.Get(keboola.ComponentID(component)); !found {
+			return false
+		}
+	}
+	return true
+}
+
+func (v *TemplateRecord) CheckFeatures(d keboola.FeaturesMap) bool {
```

Comment:
`CheckFeatures` - maybe we should rename it to `CheckProjectFeatures`, because there are also `stack features`.